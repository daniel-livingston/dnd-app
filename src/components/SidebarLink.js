import React from "react";
import { NavLink } from "react-router-dom";
import PropTypes from "prop-types";

const propTypes = {
	to: PropTypes.string.isRequired,
	exact: PropTypes.bool,
	Icon: PropTypes.elementType.isRequired,
	name: PropTypes.string.isRequired,
};

const SidebarLink = (props) => (
	<NavLink
		className='SidebarLink'
		activeClassName='SidebarLink-active'
		to={props.to}
		exact={props.exact}
	>
		<div className='SidebarLink-item'>
			<props.Icon />
			<div className='SidebarLink-title text-primary'>{props.name}</div>
		</div>
	</NavLink>
);

SidebarLink.propTypes = propTypes;

export default SidebarLink;
