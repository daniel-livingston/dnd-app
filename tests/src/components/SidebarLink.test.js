import React from "react";
import { shallow } from "enzyme";
import SidebarLink from "../../../src/components/SidebarLink";

const props = {
	to: "/test",
	Icon: () => <svg></svg>,
	name: "Test",
};

describe("<SidebarLink>", () => {
	test("renders correctly", () => {
		const wrapper = shallow(<SidebarLink {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});
