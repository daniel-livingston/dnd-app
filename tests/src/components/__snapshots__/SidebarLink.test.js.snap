// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarLink> renders correctly 1`] = `
<NavLink
  activeClassName="SidebarLink-active"
  className="SidebarLink"
  to="/test"
>
  <div
    className="SidebarLink-item"
  >
    <Icon />
    <div
      className="SidebarLink-title text-primary"
    >
      Test
    </div>
  </div>
</NavLink>
`;
